Project1/GridMap.cpp

```diff
@@ -0,0 +1,148 @@
+#include "GridMap.h"
+#include "Grid.h"
+#include "Party.h"
+#include "Unit.h"
+#include "GridCell.h"
+#include "CellPath.h"
+#include "CellEdge.h"
+#include "Mobility.h"
+#include "Passkey.h"
+
+//Constructors//
+/////////////////////////////////////////////////////////////////////////
+GridMap::GridMap(int width, int height) :
+	Grid(width, height)
+{}
+
+GridMap::GridMap(int width, int height, std::vector<PartyData> data) :
+	Grid(width, height, data)
+{}
+
+//Party Access//
+/////////////////////////////////////////////////////////////////////////
+Party& GridMap::getParty(const Unit & unit) {
+	for (auto & party : _parties) {
+		if (*unit.getParty() == party) {
+			return party;
+		}
+	}
+	Expects(false);
+}
+
+const Party& GridMap::getParty(const Unit & unit) const {
+	return unit.getParty()->getParty(Passkey<GridMap>());
+}
+
+//Non- const findShortestPaths()//
+/////////////////////////////////////////////////////////////////////////
+
+PathMap GridMap::findShortestPaths(ID start) {
+	return PathMap(*this, getShortestPaths(start, INT_MAX, MobilitySet(MobilityType::values::GROUNDED), false));
+}
+
+PathMap GridMap::findShortestPaths(size_t x, size_t y) {
+	return PathMap(*this, getShortestPaths((*this)[x][y]._id, INT_MAX, MobilitySet(MobilityType::values::GROUNDED), false));
```

Comment:
Since operator[] is public, I would remove this method and make the caller find the id they want.

And I haven't been doing this either, but [default arguments](https://www.geeksforgeeks.org/default-arguments-c/) are a thing if you want to try that out.

Reply:
Agre on this, it seems like it is needlessly verbose and useless

Reply:
check